lib/idl/main.ts

```diff
@@ -0,0 +1,55 @@
+import { Idl } from "@project-serum/anchor"
+import * as fs from "fs"
+import * as path from "path"
+import { Project } from "ts-morph"
+import { program } from "commander"
+
+import { generateAccounts } from "./accounts"
+import { generateErrors } from "./errors"
+import { generateInstructions } from "./instructions"
+import { generateProgramId } from "./programId"
+import { generateTypes } from "./types"
+
+
+
+function main() {
+  program
+    .description(
+  )
+    .argument("idl", "file path")
```

Comment:
file path is not necessary should be optional. Default behavior should be that it fetches idl from cluster.

Reply:
@godmodegalactus , Done ! The recent implementation takes address and output directory as arguments and file path as options.